collect_reprogs.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+
+import argparse
+import glob
+import filecmp
+import logging
+import os
+import shutil
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Collect C repros programs from subdirs.')
+    parser.add_argument('directory', type=str,
+                        help="Folder to be searched for reprogs.")
+    parser.add_argument('--output', '-o', required=False, default="./linux",
+                        help="Output folder where to store. default ./linux")
+    args = parser.parse_args()
+
+    workdir = os.path.join(args.directory, "")
+    logging.debug(workdir)
+
+    for file in glob.iglob(workdir + '**/*.cprog', recursive=True):
+        logging.debug(file)
+        file_path = file.split('/')
+        has_id_index = len(file_path) - 2
+
+        candidate_name = file_path[has_id_index]
+        candidate_file_name = os.path.join(args.output, candidate_name + ".c")
+        index = 1
+        already_exist = False
+        while os.path.exists(candidate_file_name):
+            if filecmp.cmp(candidate_file_name, file, shallow=False):
+                already_exist = True
+                break
+            candidate_file_name = os.path.join(args.output, candidate_name + "_" + str(index).rjust(3, '0') + ".c")
```

Comment:
What we currently do in reprolist.go and what's in the current corpus and what Shuah Khan asked for is 1 reproducer per bug. And syz-manager should also generally create no more than 1. Let's keep it consistent and do 1 repro per bug without suffix.

However, looking at your repros, I see you have up to 7 repros per bug. How did it happen? syz-manager should not create than many. Some also have links to syzkaller.appspot.com, which is strange... did you saved then from syzbot dashboard?

Reply:
Hmm. It turned out that my colleague had copied the reproducers from the syzkaller dashboard when he was debugging one bug. He told that he downloaded those from the dashboard.

I will change the script to save only one instance by default.